rct229/rulesets/ashrae9012019/section11/section11rule10.py

```diff
@@ -0,0 +1,508 @@
+                        # Note: Draw pattern is not used in this lookup
+                        efficiency_data = table_7_8_lookup(
+                            "Electric storage water heater",
+                            swh_input_power_b,
+                        )
+
+                # Get Gas Storage Water Heater Efficiency Data
+                elif (
+                    swh_fuel_type_b
+                    in [
+                        EnergySourceOptions.NATURAL_GAS,
+                        EnergySourceOptions.PROPANE,
+                    ]
+                    and swh_tank_type_b in STORAGE_TYPES
+                ):
+                    if swh_input_power_b <= 75000 * ureg("Btu/h"):
+                        # Lookup the expected efficiency from Appendix F-2
+                        assert_(
+                            draw_pattern_b is not None,
+                            "Draw pattern must be defined for table F-2 lookup",
+                        )
+                        efficiency_data = table_f_2_lookup(
+                            "Gas storage water heater",
+                            swh_tank_storage_volume_b,
+                            draw_pattern_b,
+                        )
+                    else:
+                        # Lookup the expected efficiency from Table 7.8
+                        assert_(
+                            draw_pattern_b is not None,
+                            "Draw pattern must be defined for table 7-8 lookup",
+                        )
+                        efficiency_data = table_7_8_lookup(
+                            "Gas storage water heater",
+                            swh_input_power_b,
+                            draw_pattern_b,
+                        )
+                        # Note: If the input power is greater than 105,000 Btu/h, there will be a thermal efficiency and standby loss target
+
+                if efficiency_data:
+                    to_remove = None
+                    for efficiency in efficiency_data:
+                        if "STANDBY" in efficiency["metric"]:
+                            standby_loss_target_b = eval(
+                                efficiency["equation"],
+                                {"__builtins__": None},
+                                {
+                                    "v": swh_tank_storage_volume_b.to(
+                                        "gallon"
+                                    ).magnitude,
+                                    "q": swh_input_power_b.to("Btu/h").magnitude,
+                                },
+                            )
+                            standby_loss_target_metric_b = efficiency["metric"]
+                            modeled_standby_loss_b = swh_efficiency_b.get(
+                                standby_loss_target_metric_b
+                            )
+                            to_remove = efficiency
```

Comment:
I dont quite understand the `remove` here.
The current implement removes the first one meeting the condition, and assign `modeled_standby_loss_b`
Then at line 248, the code requires only one `efficiency_data` remain for `modeled_efficiency_b`

In scenarios 
1. one matched entries - this could return either `modeled_standby_lss_b` is None and `modeled_efficiency_loss_b` not None or `modeled_standby_loss_b` is not None and `modeled_efficiency_loss_b` is None.
2. Two matched entries - depends on condition,`modeled_standby_loss_b` and `modeled_efficiency_b` could be both None or have some values.
3. More than three matched entries, than this will only return `modeled_standby_loss_b` or both `modeled_standby_loss_b` and `modeled_efficiency_b` are None.

Is the bullet point 3 a desired behavior? If bullet point 3 condition would never met, than clearly there is pre-condition for the efficiency_data list.

Reply:
The length of efficiency_data will be a maximum length of 2 based on the usage of the table lookups. This will only occur in the case where it is a gas storage water heater with input rating >105,000 Btu/h where it will include the efficiency rating 80% Et and also the standby loss rating SL >= (Q/800 + 110 ), Btu/h

Reply:
i see, that is the precondition for the `efficiency_data`. Could you add a comment to describe this pre_condition?